### PR TITLE
FLUID-5846: Fixed tests and implementation of tooltips to cooperate with iframes

### PR DESCRIPTION
**Pull request by amb26**

### CODE REVIEW

File: src/framework/core/js/FluidView.js [hunk 1]
```diff
@@ -448,6 +448,17 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
...
 
     /**
+     * Returns the document to which an element belongs, or the element itself if it is already a document
+     *
+     * @param @param {jQuery||Element} element the element to return the document for
+     * @return {Document} dokkument the document in which it is to be found
+     */
+    fluid.getDocument = function (element) {
+        var node = fluid.unwrap(element);
+        return node.nodeType === 9 ? node : node.ownerDocument;
```
Comment (jobara) [inline src/framework/core/js/FluidView.js]: I'm assuming nodeType of 9 is a document. Could you leave a comment here about that?

File: src/framework/core/js/FluidView.js [hunk 2]
```diff
@@ -448,6 +448,17 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
     };
 
     /**
+     * Returns the document to which an element belongs, or the element itself if it is already a document
+     *
+     * @param @param {jQuery||Element} element the element to return the document for
+     * @return {Document} dokkument the document in which it is to be found
+     */
```
Comment (jobara) [inline src/framework/core/js/FluidView.js]: Is it useful to add documentation about this at http://docs.fluidproject.org/infusion/development/ViewAPI.html

File: tests/component-tests/tooltip/js/TooltipTests.js [hunk 3]
```diff
@@ -194,10 +194,107 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
...
+            iframe: ".FLUID-5846-iframe"
+        },
+        events: {
+            iframeReady: null,
+            onReady: null,
+            afterOpen: null,
+            afterClose: null
+        },
+        listeners: {
+            "onCreate.setupIframe": "fluid.tests.tooltip.FLUID5846.setupIframe({that}, {that}.dom.iframe.0)"
```
Comment (jobara) [inline tests/component-tests/tooltip/js/TooltipTests.js]: Should `{that}.dom.iframe.0` just be `{that}.dom.iframe` as the function seems to need to use it as a jQuery element anyways.

Comment (amb26) [inline tests/component-tests/tooltip/js/TooltipTests.js]: Line 217 uses the plain DOM node value

File: src/framework/core/js/FluidView.js (hunk 2 above)
Comment (amb26) [inline src/framework/core/js/FluidView.js]: Matching docs pull here: https://github.com/fluid-project/infusion-docs/pull/82

**Comment by jobara** [comment]
merged at ab20b889330ee0b8eb37471fb166e315419f8e12

**Pull request merged**